src/core/variable_mx.jl

```diff
@@ -0,0 +1,344 @@
+"""
+function variable_mx_complex(model::JuMP.Model, indices::Array{T,1}, N::Int, M::Int;
+        upper_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing, lower_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing,
+        symm_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing,
+        name::Union{String, Tuple{String,String}}="", prefix="") where T
+
+    if !ismissing(symm_bound)
+        @assert(ismissing(upper_bound) && ismissing(lower_bound), "When a symmetric bound is specified, no lower or upper bound can be specified.")
+        upper_bound = symm_bound
+        lower_bound = Dict{T, Array{Real,2}}([(k,-v) for (k,v) in symm_bound])
+    end
+
+    name_real = isa(name, Tuple) ? name[1] : "$(name)r"
+    name_imag = isa(name, Tuple) ? name[2] : "$(name)i"
+    Mre = variable_mx_real(model, indices, N, M;
+        upper_bound=upper_bound, lower_bound=lower_bound,
+        prefix=prefix, name=name_real)
+    Mim = variable_mx_real(model, indices, N, M;
+        upper_bound=upper_bound, lower_bound=lower_bound,
+        prefix=prefix, name=name_imag)
+    return (Mre,Mim)
+end
+
+
+"""
+Same as variable_mx_complex, but square and the diagonal of the matrix variables
+consists of the constants passed as the diag_re and diag_im argument. The diag
+argument is a dictionary of (index, 1d-array) pairs.
+Useful for power matrices with specified diagonals (constant power wye loads).
+"""
+function variable_mx_complex_with_diag(model::JuMP.Model, indices::Array{T,1}, N::Int;
+        upper_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing, lower_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing,
+        symm_bound::Union{Missing, Dict{T,<:Array{<:Real,2}}}=missing,
+        diag_re::Dict{T,<:Array{<:Any,1}}=Dict([(i, zeros(N)) for i in indices]),
+        diag_im::Dict{T,<:Array{<:Any,1}}=Dict([(i, zeros(N)) for i in indices]),
+        name::Union{String, Tuple{String,String}}="", prefix="") where T
+
+    if !ismissing(symm_bound)
+        @assert(ismissing(upper_bound) && ismissing(lower_bound), "When a symmetric bound is specified, no lower or upper bound can be specified.")
+        upper_bound = symm_bound
+        lower_bound = Dict([(k,-v) for (k,v) in symm_bound])
+    end
+
+    name_real = isa(name, Tuple) ? name[1] : "$(name)r"
+    name_imag = isa(name, Tuple) ? name[2] : "$(name)i"
+    Mre = variable_mx_real_with_diag(model, indices, N;
+        upper_bound=upper_bound, lower_bound=lower_bound,
+        diag=diag_re, prefix=prefix, name=name_real
+    )
+    Mim = variable_mx_real_with_diag(model, indices, N;
+        upper_bound=upper_bound, lower_bound=lower_bound,
+        diag=diag_im, prefix=prefix, name=name_imag)
+    return (Mre,Mim)
+end
+
+
+# HERMITIAN MATRICES
+#%##########################
+
+
+"""
+Same as variable_mx_real, but adds symmetry structure
+"""
+function variable_mx_symmetric(model::JuMP.Model, indices::Array{T,1}, N::Int;
```

Comment:
maybe name should be `variable_mx_real_symmetric` to better indicate it is directly related to `variable_mx_real`